packages/build-utils/test/unit.detect-package-manager.test.ts

```diff
@@ -87,6 +90,26 @@ describe('Test `detectPackageManager()`', () => {
           path: '/pnpm9/node_modules/.bin',
         },
       },
+      {
+        name: 'for 9.0 lockfile returns pnpm 9 path before prefer pnpm 10 datetime',
+        args: ['pnpm', 9.0, PNPM_10_PREFERRED_AT.getTime() - 1000],
+        want: {
+          detectedLockfile: 'pnpm-lock.yaml',
+          detectedPackageManager: 'pnpm@9.x',
+          pnpmVersionRange: '9.x',
+          path: '/pnpm9/node_modules/.bin',
+        },
+      },
+      {
+        name: 'for 9.0 lockfile returns pnpm 10 path after prefer pnpm 10 datetime',
+        args: ['pnpm', 9.0, PNPM_10_PREFERRED_AT.getTime() * 1000],
```

Comment:
```suggestion
        args: ['pnpm', 9.0, PNPM_10_PREFERRED_AT.getTime() + 1000],
```

This works, but I think you meant to use `+` here.

Reply:
Whoops, thanks for catching that!